Engine/Generic/SuppressedRecord.cs

```diff
@@ -24,6 +24,7 @@ public class SuppressedRecord : DiagnosticRecord
         public SuppressedRecord(DiagnosticRecord record, IReadOnlyList<RuleSuppression> suppressions)
         {
             Suppression = new ReadOnlyCollection<RuleSuppression>(new List<RuleSuppression>(suppressions));
+            IsSuppressed = true;
```

Comment:
Could this ever be called with a suppression list count of zero? In that case IsSuppressed really true?

Reply:
Very good question. I think the mere wrapping in the `SuppressedRecord` object classes the diagnostic as suppressed, but I agree that this opens up the possibility of an invalid object, which I dislike... I'm not sure I see a way around this without changing a few things around though, and I'm not sure that's worth it.